### ISSUE TITLE
Seams and overdraw visible with transparent raster tiles + globe view

### ISSUE THREAD

**Issue by rreusser** [mapbox/mapbox-gl-js#11906]
**mapbox-gl-js version**: current `main`

**browser**: Chrome 101.0.4951.64 

### Steps to Trigger Behavior

 1. Load a tileset with alpha. One like the following set of tiles (to be placed in `debug/transparent-white`) will do.
 2. View a map with `projection: {name: 'globe'}`
 3. Zoom and pan

A single 50% opacity white tile:
![0](https://user-images.githubusercontent.com/572717/169118877-9de180fd-536a-4749-a3d4-314fae148ebe.png)

A tileset of the above, through z=4:
[transparent-white.zip](https://github.com/mapbox/mapbox-gl-js/files/8719292/transparent-white.zip)

<details>
<summary>debug/testpage.html</summary>

  ```html
    <!DOCTYPE html>
    <html>
    <head>
        <title>Mapbox GL JS debug page</title>
        <meta charset='utf-8'>
        <meta name='viewport' content='width=device-width, initial-scale=1.0, user-scalable=no'>
        <link rel='stylesheet' href='../dist/mapbox-gl.css' />
        <style>
            body { margin: 0; padding: 0; }
            html, body, #map { height: 100%; }
        </style>
    </head>
    <body>
    <div id="map"></div>
    <script src='../dist/mapbox-gl-dev.js'></script>
    <script src='../debug/access_token_generated.js'></script>
    <script>

    var map = window.map = new mapboxgl.Map({
        container: 'map',
        style: 'mapbox://styles/mapbox/satellite-v9',
        hash: true,
        projection: { name: 'globe' }
    });

    map.on('load', function () {
        map.addSource('transparent-white', {
            type: 'raster',
            minzoom: 0,
            maxzoom: 4,
            tileSize: 512,
            tiles: [
                'http://localhost:9966/debug/transparent-white/{z}/{x}/{y}.png'
            ]
        });

        map.addLayer({
            type: 'raster',
            source: 'transparent-white',
            id: 'transparent-white-layer',
        });
    });

    </script>
    </body>
    </html>
  ```
</details>

### Expected Behavior

A tileset with opacity is rendered correctly—or even not at all, but preferably not incorrectly.

### Actual Behavior

![Screen Shot 2022-05-18 at 11 27 20 AM](https://user-images.githubusercontent.com/572717/169121556-b9629336-d40f-4eb7-9a58-ebc71759efa1.png)

There are two problems:
- seams between tiles are visible
- when zooming out, sometimes multiple tiles are overdrawn—usually toward the edges—such that 50% opacity + 50% opacity ~= 75% opacity. This does not resolve when the map idles.

In my testing, I obtain the expected results on 2D maps, even with projections. It's the globe view where this seems to occur. It seems like a stencil test might be able to prevent this sort of overdraw?

**Comment by rreusser** [comment]
Although I should note that if I understand thing correctly, while the web is fine, maybe it's best not to rely on transparent tilesets in general. If a platform didn't load webp tiles and png tiles were too big to load, then it'd receive jpeg tiles and transparency wouldn't be possible at all.

**Comment by rreusser** [comment]
The seams likely stem from the epsilon-overlap in this PR: https://github.com/mapbox/mapbox-gl-js/issues/11858

**Comment by SnailBones** [comment]
@rreusser That was my thinking too, but it seems the [fix](https://github.com/mapbox/mapbox-gl-js/pull/11871) to that issue hasn't been merged yet.

**Comment by karimnaaji** [comment]
>@rreusser That was my thinking too, but it seems the https://github.com/mapbox/mapbox-gl-js/pull/11871 to that issue hasn't been merged yet.

Yes I wasn't really happy with the result of https://github.com/mapbox/mapbox-gl-js/pull/11871, so I left it as an experiment. The issue with adding some extra padding in the proposed PR is that it trades it off for other issues such as overlapping artifacts (such as z-fighting or when using transparent tiles an incorrect transparent contribution). One other potential fix would be to draw the background color in the space that's filled by the footprint of the globe, that would get rid of these artifacts but might not work great with transparency.

**Comment by rreusser** [comment]
@karimnaaji two tricks I'm wondering about:
1. How does stencil clipping work? I never wrapped my head around it. We draw low-zoom (or high-zoom?) tiles first and use stencil to ensure they overlap correctly? Almost like a depth buffer? I can't quite work out in my head how that would work with the tile skirt, for example. Would this enable anything to prevent drawing the same transparent tile on top of already-rendered transparent tiles of different zoom levels—which I think is where the issue is coming from, right?
2. Would polygon offset offer anything? As long as meshes are exactly identical, this would prevent z-fighting. I think the problem though is for tiles of different zoom levels or for differently-meshed tiles meeting, so that I don't think this would offer anything.

**Comment by rreusser** [comment]
Sorry these are half-formed thoughts, but I'm thinking along the lines of this: https://github.com/mapbox/mapbox-gl-js/pull/11546 Does this relate to the bug or perhaps offer a solution?

**Comment by RomanAtBrief** [comment]
Was it fixed ? Any updates when or workarounds. I am still getting this problem with GL JS

**Comment by RomanAtBrief** [comment]
Just to add, the same issue is with any raster layer, not only satellite image
